src/org/parosproxy/paros/extension/state/ExtensionState.java

```diff
@@ -67,11 +75,46 @@ public void hook(ExtensionHook extensionHook) {
 	    // ZAP: Changed to check if there is a view.
 	    if (getView() != null) {
 		    extensionHook.getHookMenu().addEditMenuItem(extensionHook.getHookMenu().getMenuSeparator());
+		    extensionHook.getHookMenu().addEditMenuItem(getMenuZAPMode());
 		    extensionHook.getHookMenu().addEditMenuItem(getMenuSessionTrackingEnable());
 		    extensionHook.getHookMenu().addEditMenuItem(getMenuResetSessionState());
 	    }
 	}
 
+	private JMenuItem getMenuZAPMode(){
+		if (menuZAPMode == null) {
+			menuZAPMode = new JMenu(Constant.messages.getString("menu.edit.zapmode"));
+			int modeCount = 0;
+			for (Mode modeType : Mode.values()) {
+			menuItemsMode[modeCount] = addZAPModeMenuItem(modeType);
+  		menuZAPMode.add(menuItemsMode[modeCount]);
+			modeCount++;
+			}
+			modeCount = 0;
+			for(Mode modeType : Mode.values()){
+			addZAPModeEventListeners(modeType,menuItemsMode[modeCount],menuItemsMode);

```

Comment:
I need all elements of my array of checkboxes i.e. menuItemsMode[] to be set before I pass the array as an argument to my addZAPModeEventListeners() function. This is because whenever I click on either of the modes, I am first deselecting all the checkboxes and then checking the correct one. So, I need all my menu items to be there before I add any eventListener and hence, before adding event listeners, I am first adding all the checkbox items and then I am looping again to add eventListeners. It can be done easily with a radio button type of menu but then the GUI would look different compared to other menu items in ZAP.